openquake/hazardlib/contexts.py

```diff
@@ -185,9 +187,14 @@ def make_contexts(self, sites, rupture):
         sites, dctx = self.filter(sites, rupture)
         for param in self.REQUIRES_DISTANCES - set([self.filter_distance]):
             distances = get_distances(rupture, sites, param)
-            if param == 'repi' and self.reqv:
-                distances = self.reqv.get(distances, rupture.mag)
             setattr(dctx, param, distances)
+        if self.reqv and reqv_point:
+            reqv = self.reqv.get(getattr(dctx, 'repi'), rupture.mag)
+            if 'rjb' in self.REQUIRES_DISTANCES:
+                setattr(dctx, 'rjb', reqv)
+            if 'rrup' in self.REQUIRES_DISTANCES:
+                reqv_rup = numpy.sqrt(reqv**2 + rupture.hypocenter.depth**2)
+                setattr(dctx, 'rrup', reqv_rup)
```

Comment:
You can just write `dctx.rrup = reqv_rup`

Reply:
ok, will do